frontend/tests/unit/inst_pointer_chase_gen_test.py

```diff
@@ -120,5 +121,52 @@ def test_generate_cfg(self):
         self.assertEqual(len(cfg.code_block_bodies), num_total_functions + 1)
 
 
+class CodePrefetchInstPointerChaseGeneratorTest(unittest.TestCase):
+
+    def setUp(self):
+        self.depth = 3
+        self.num_callchains = 2
+        self.gen = inst_pointer_chase_gen.InstPointerChaseGenerator(
+            self.depth,
+            self.num_callchains,
+            True,  # insert_code_prefetches
+            function_selector=_pop_next_function)
+
+    def test_add_code_prefetch_code_block(self):
```

Comment:
Do we want to add a test for when both function_id and code_block_id is provided?

Reply:
Done.